### PR TITLE
Quantization docs

### PR DESCRIPTION
**Pull request by angelayi**
Mentally stacked on top of https://github.com/pytorch/pytorch/pull/100577 (will land that one first).

### CODE REVIEW

**Review by msaroufim — APPROVED** [review]

File: docs/source/compile/quantization.rst [hunk 1]
```diff
@@ -0,0 +1,147 @@
+(prototype) Quantization with PT2
```
Comment (msaroufim) [inline docs/source/compile/quantization.rst:1]: One thing I'm not super clear on is what dtypes are actually supported? Is it CPU? GPU? Both?

**Comment by angelayi** [comment]
Converting to draft because quantization isn't entirely ready to make the APIs public yet. Will publish this once they are ready!

File: docs/source/compile/quantization.rst [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+=================================
+
+This document will discuss the extension of Quantization-Aware Training
+(QAT) with PT2. To learn about quantization in general, please visit
```
Comment (leslie-fang-intel) [inline docs/source/compile/quantization.rst:5]: Looks like both PTQ and QAT is discussed in this doc?

File: docs/source/compile/quantization.rst [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+           """
+           Validate the annotated graph is supported by the backend
+           """
+           ...
+
+       # annotate nodes in the graph with observer or fake quant constructors
+       # to convey the desired way of quantization
+       def get_supported_operators(cls) -> List[OperatorConfig]:
+           """
+           Return a list of ATen operators, `torch.nn.Module` types, or
```
Comment (leslie-fang-intel) [inline docs/source/compile/quantization.rst:141]: nit, looks like ATen operators is missing (only `torch.nn.Module` types and `torch.nn.functional` operators are included) from current `QNNPackQuantizer` example https://github.com/pytorch/pytorch/blob/35a6b04419133e5f7bdaaed8cf1c744e340456b3/torch/ao/quantization/_pt2e/quantizer/qnnpack_quantizer.py#L92-L97

File: docs/source/compile/quantization.rst (hunk 2 above)
Comment (leslie-fang-intel) [reply docs/source/compile/quantization.rst:5]: What's about PTQ dynamic quant? Should we also provide some examples here?

**Pull request closed without merging**